samples/WebFormsToBlazor/BlazorCore/Program.cs

```diff
@@ -1,27 +1,62 @@
 using BlazorCore;
-using BlazorCore.Data;
+using BlazorCore.Components;
 using Microsoft.AspNetCore.Components.Web;
 
 var builder = WebApplication.CreateBuilder(args);
 
-builder.Services.AddSystemWebAdapters();
-builder.Services.AddReverseProxy();
+// Add YARP
+builder.Services.AddHttpForwarder();
+
+// Add constraint to redirect .axd files to WebForms
+builder.Services.AddRouting(options => options.ConstraintMap.Add("isAxdFile", typeof(AxdConstraint)));
+// Add System Web Adapters and setup session
+builder.Services.AddSystemWebAdapters()
+    .AddJsonSessionSerializer(options =>
+    {
+        options.RegisterKey<string>("test-value");
+    })
+    .AddRemoteAppClient(options =>
+    {
+        options.RemoteAppUrl = new(builder.Configuration["ProxyTo"]);
+        options.ApiKey = builder.Configuration["RemoteAppApiKey"];
+    })
+    .AddSessionClient();
 
 // Add services to the container.
-builder.Services.AddRazorPages();
-builder.Services.AddServerSideBlazor(options =>
-{
-    options.RootComponents.RegisterCustomElement<HelloWorld>("hello-world");
-});
-builder.Services.AddSingleton<WeatherForecastService>();
+builder.Services.AddRazorComponents()
+    .AddInteractiveServerComponents(options => options.RootComponents.RegisterCustomElement<HelloWorld>("hello-world"));
 
 var app = builder.Build();
 
+app.UseHttpsRedirection();
+
 app.UseStaticFiles();
-app.UseRouting();
+app.UseAntiforgery();
+
+/*
+ * When Blazor interactive SSR establishes a SignalR connection, a CONNECT method is requested.
+ * This request will stay open until Blazor sends a disconnect request. 
+ * Since it stays open, the session is not properly committed and will stay in the heartbeat loop on the framework side of System.WebAdapters.
+ * We can either:
+ *     A: Block all connect requests from using System.Web.Adapters
+ *     B: Block all requests from paths containing "/_blazor"
+ * TLDR: We need to ensure that SystemWeb.Adapters is not used with Blazor SignalR
+ */
+app.UseWhen(
```

Comment:
Is there a tracking issue for this? Not sure if this is the best way, but we should have an out of the box way to handle this I think.

Reply:
Not that I know of. I mentioned the deadlock in my previous PR and ended up with this workaround to get my project working.

Reply:
sounds good for the sample for now. could you create an issue to track this?

Reply:
#543